### ISSUE TITLE
[Wii U] Erratic input management on most cores

### ISSUE THREAD

**Issue by MeisterLLD** [libretro/RetroArch#14935]
## Description
On most cores (everyone I've tried besides FB Alpha and Mame), input management is very erratic on Wii U.

- the only working controller is the Wii U Gamepad, and only if it is set as port 1 input
- even when "all users control menu" is on, it does not work most of the time. For instance if my Wii U Pro Controller is connected before opening retroarch, and even if I set it as port 2 input, it can't control the menu. If I connect it after opening Retroarch, it can, until I change inputs, then even if I come back to the Wii U Pro Controller, it can't control the menu anymore. 
- The Gamecube Controller (Mayflash USB adapter) seems to consistently be able to control the menu though
- If I set something other than the Wii U Gamepad as port 1 input, then I can't control anything anymore, sometimes not even the menu (I mean : even the Wii U Gamepad can't control the menu anymore)
- Weirdly enough, hotkeys work fine in game. E.g. if I set the Wii U Pro controller as port 1 input and bind some hotkeys on it, game inputs are not recorded but hotkeys are.

### Expected behavior

- Controllers set up as port 1 input should be able to control the game
- All controllers connected and setup as port x input for some x, should be able to control the menu, once "all users control menu" is setup.

### Actual behavior

See description.

### Steps to reproduce the bug

1. Open RetroArch, change port 1 input from Wii U Gamepad to something else

### Bisect Results

Happened since as long as I can remember, even though I haven't been testing intermediate builds since around a year ago and until now.

### Version/Commit
Latest nightlies.

- RetroArch:  latest nightly.

### Environment information
Wii U.

**Comment by MeisterLLD** [comment]
It might be interesting to note that FB Alpha and Mame work fine, and those setup initially Wii U Gamepad to "port #0". It also seems to me that there is something fishy about a +1 translation, since I have the impression that port #2 controls the menu in other cores. I've just found out that if switch port 2 from, say, Wii U Pro Controller, which can control the menu just before that, then it can't anymore.

**Comment by MeisterLLD** [comment]
Good temporary fix : use core settings to remap Port 2 to Player 1 (and Port 3 to Player 2 etc). This works flawlessly. But the problem is that then one cannot use hotkeys anymore...

**Comment by Gregoloco** [comment]
Hello. Did anyone find a solution to this ?
I would love so much to use RetroArch on my WiiU but I cannot play with my pro controllers only and this ruins everything... I would love so much to be able to set the Wii u gamepad to Port 5 and the pro controllers to ports 1 to 4 (with the home BTN working on pro controller).
Please, I'm waiting for this for years now and it would really be a dream come true 😘

**Comment by MeisterLLD** [comment]
Yes. What I've written just above your post works perfectly. Except for hotkeys... Home button works.

**Comment by Gregoloco** [comment]
Cool I have to try.
Should I remap ports for every cores one by one though ?

**Comment by Gregoloco** [comment]
Sorry to disturb, but I really can't find where to remap ports for each core

**Comment by Gregoloco** [comment]
Thanks so much man.
It totally worked. I'm so happy.
Thanks again ☺️
